internal/service/atlasuser/data_source_atlas_users_test.go

```diff
@@ -223,8 +223,8 @@ func dataSourceChecksForUsers(dataSourceName, orgID string, users *admin.Paginat
 		resource.TestCheckResourceAttr(dataSourceName, "org_id", orgID),
 		resource.TestCheckResourceAttr(dataSourceName, "total_count", fmt.Sprintf("%d", totalCountValue)),
 	}
-	for i := range users.Results {
-		checks = append(checks, dataSourceChecksForUser(dataSourceName, fmt.Sprintf("results.%d.", i), &users.Results[i])...)
+	for i := range *users.Results {
+		checks = append(checks, dataSourceChecksForUser(dataSourceName, fmt.Sprintf("results.%d.", i), &(*users.Results)[i])...)
```

Comment:
wouldn't this be the same as
```suggestion
		checks = append(checks, dataSourceChecksForUser(dataSourceName, fmt.Sprintf("results.%d.", i), users.Results)[i])...)
```

Reply:
doing users.Results[i] gives this error:  "invalid operation: cannot index users.Results (variable of type *[]"go.mongodb.org/atlas-sdk/v20231115003/admin".CloudAppUser)"

Reply:
what if you use the getter?

Reply:
this works: &users.GetResults()[i]
changed